ark/phenotyping/som_utils.py

```diff
@@ -80,46 +83,96 @@ def create_pixel_matrix(img_xr, seg_labels, fovs=None, channels=None, blur_facto
         pixel_mat.loc[:, channels] = pixel_mat.loc[:, channels].div(
             pixel_mat.loc[:, channels].sum(axis=1), axis=0)
 
-        # assign to flowsom_data if not already assigned, otherwise concatenates
-        if flowsom_data is None:
-            flowsom_data = pixel_mat
-        else:
-            flowsom_data = pd.concat([flowsom_data, pixel_mat])
+        # write dataset to hdf5 file, mode 'a' will create if it doesn't exist
+        pixel_mat.to_hdf(os.path.join(base_dir, hdf_name), key=fov, mode='a')
+
+
+def subset_pixels(fovs, base_dir, hdf_name='pixel_mat_preprocessed.hdf5',
+                  csv_name='pixel_mat_subsetted.csv', subset_percent=0.1):
+    """Takes a random percentage subset of pixel data from each fov
+
+    Args:
+        fovs (list):
+            The list of fovs to read
+        base_dir (str):
+            Name of the directory to save the subsetted CSV to
+        hdf_name (str):
+            Name of the file which contains the preprocessed pixel data,
+            defaults to pixel_mat_preprocessed.hdf5
+        csv_name (str):
+            Name of the file to write the subsetted pixel data frame to
+        subset_percent (float):
+            The percentage of pixels to take from each fov, defaults to 0.1
+    """
```

Comment:
I feel like this can be merged with the `create_pixel_mat` function.  Also, any reason the subset isn't and hdf5?

Reply:
I've been thinking about that. And yes, the subset should be HDF5, I just got lazy cause CSVs are easier to work with 😅.

Reply:
I agree, lets add this as a couple lines at the end of ```create_pixel_mat```